src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -1512,6 +1512,21 @@ void GenericThreadStackManagerImpl_OpenThread<ImplClass>::OnSrpClientNotificatio
     case OT_ERROR_NONE: {
         ChipLogProgress(DeviceLayer, "OnSrpClientNotification: Last requested operation completed successfully");
 
+        if (aHostInfo)
+        {
+            if (aHostInfo->mState == OT_SRP_CLIENT_ITEM_STATE_REMOVED)
+            {
+                GenericThreadStackManagerImpl_OpenThread * threadStackInstance = static_cast<GenericThreadStackManagerImpl_OpenThread *>(aContext);
+
+                // Clear memory for removed host
+                char * hostname = threadStackInstance->Impl()->mSrpClient.mHostName;
+                memset(reinterpret_cast<void *>(hostname), 0, sizeof(hostname));
```

Comment:
That's wrong. `sizeof(hostname) != sizeof(threadStackInstance->Impl()->mSrpClient.mHostName)`. You can define `hostname` as `auto&` which will mean a reference to the array. Also, the `reinterpret_cast` is unnecessary.